pageserver/src/tenant/timeline.rs

```diff
@@ -2282,14 +2285,17 @@ impl Timeline {
         // accurate relation sizes, and they do not emit consumption metrics.
         debug_assert!(self.tenant_shard_id.is_zero());
 
-        let _guard = self.gate.enter();
+        let guard = self
+            .gate
+            .enter()
+            .map_err(|_| CalculateLogicalSizeError::Cancelled)?;
```

Comment:
This was the missed Result. Unlikely this would had been hit except for attach right before shutdown.